src/pytype.jl

```diff
@@ -280,7 +280,7 @@ type PyTypeObject
              Py_TPFLAGS_HAVE_CLASS |
              Py_TPFLAGS_HAVE_STACKLESS_EXTENSION |
              Py_TPFLAGS_HAVE_INDEX)
-        name_save = bytestring(name)
+        name_save = Base.cconvert(Ptr{UInt8}, name)
```

Comment:
Should be `String(name)`.   The `name` argument is an `AbstractString`, and could potentially be some other string type like `UTF16String`, and needs to be converted to UTF-8.